### ISSUE TITLE
No rule to make target 'zkllvm_zkllvm'.

### ISSUE THREAD

**Issue by BoltonBailey** [NilFoundation/zkllvm-template#12]
Hello,

I have followed the instructions to try to get this working. When I get to the `cmake .. && make zkllvm_zkllvm` instruction I get the following error:
```
root@41d4bdedd740:/opt/zkllvm-template# mkdir build && cd build
root@41d4bdedd740:/opt/zkllvm-template/build# cmake .. && make zkllvm_zkllvm
CMake Warning (dev) in CMakeLists.txt:
  No project() command is present.  The top-level CMakeLists.txt file must
  contain a literal, direct call to the project() command.  Add a line of
  code such as

    project(ProjectName)

  near the top of the file, but after cmake_minimum_required().

  CMake is pretending there is a "project(Project)" command on the first
  line.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- The C compiler identification is GNU 11.3.0
-- The CXX compiler identification is Clang 16.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.74.0/BoostConfig.cmake (found version "1.74.0") found components: container random system unit_test_framework 
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Looking for clock_gettime
-- Looking for clock_gettime - not found
-- Looking for gettimeofday
-- Looking for gettimeofday - found
-- Could not find support for SSE2 on this machine.
-- Could not find support for SSE3 on this machine.
-- Could not find support for SSSE3 on this machine.
-- Could not find support for SSE4.1 on this machine.
-- Could not find support for SSE4.2 on this machine.
-- Performing Test C_HAS_AVX_1
-- Performing Test C_HAS_AVX_1 - Failed
-- Performing Test C_HAS_AVX_2
-- Performing Test C_HAS_AVX_2 - Success
-- Performing Test C_HAS_AVX2_1
-- Performing Test C_HAS_AVX2_1 - Failed
-- Performing Test C_HAS_AVX2_2
-- Performing Test C_HAS_AVX2_2 - Success
-- Performing Test CXX_HAS_AVX_1
-- Performing Test CXX_HAS_AVX_1 - Failed
-- Performing Test CXX_HAS_AVX_2
-- Performing Test CXX_HAS_AVX_2 - Success
-- Performing Test CXX_HAS_AVX2_1
-- Performing Test CXX_HAS_AVX2_1 - Failed
-- Performing Test CXX_HAS_AVX2_2
-- Performing Test CXX_HAS_AVX2_2 - Success
CMake Deprecation Warning at libs/crypto3/libs/containers/CMakeLists.txt:32 (cmake_policy):
  The OLD behavior for policy CMP0079 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- Could not find support for SSE2 on this machine.
-- Could not find support for SSE3 on this machine.
-- Could not find support for SSSE3 on this machine.
-- Could not find support for SSE4.1 on this machine.
-- Could not find support for SSE4.2 on this machine.
-- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.74.0/BoostConfig.cmake (found version "1.74.0")  
-- Could not find support for SSE2 on this machine.
-- Could not find support for SSE3 on this machine.
-- Could not find support for SSSE3 on this machine.
-- Could not find support for SSE4.1 on this machine.
-- Could not find support for SSE4.2 on this machine.
-- Could not find support for SSE2 on this machine.
-- Could not find support for SSE3 on this machine.
-- Could not find support for SSSE3 on this machine.
-- Could not find support for SSE4.1 on this machine.
-- Could not find support for SSE4.2 on this machine.
-- Could NOT find TomMath (missing: TomMath_LIBRARY TomMath_INCLUDE_DIRS) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Configuring done
-- Generating done
-- Build files have been written to: /opt/zkllvm-template/build
make: *** No rule to make target 'zkllvm_zkllvm'.  Stop.
root@41d4bdedd740:/opt/zkllvm-template/build# ls
CMakeCache.txt  CMakeFiles  CTestTestfile.cmake  Makefile  arch.c  cmake_install.cmake  libs  src
root@41d4bdedd740:/opt/zkllvm-template/build# 
```

**Comment by nkaskov** [comment]
Fixed by #11.

**Comment by BoltonBailey** [comment]
Ok, but I still don't see a `zkllvm_zkllvm.ll` file

```
root@e6548116753f:/opt/zkllvm-template/build# cmake .. && make template
CMake Warning (dev) in CMakeLists.txt:
  No project() command is present.  The top-level CMakeLists.txt file must
  contain a literal, direct call to the project() command.  Add a line of
  code such as

    project(ProjectName)

  near the top of the file, but after cmake_minimum_required().

  CMake is pretending there is a "project(Project)" command on the first
  line.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.74.0/BoostConfig.cmake (found version "1.74.0") found components: container random system unit_test_framework 
-- Could not find support for SSE2 on this machine.
-- Could not find support for SSE3 on this machine.
-- Could not find support for SSSE3 on this machine.
-- Could not find support for SSE4.1 on this machine.
-- Could not find support for SSE4.2 on this machine.
CMake Deprecation Warning at libs/crypto3/libs/containers/CMakeLists.txt:32 (cmake_policy):
  The OLD behavior for policy CMP0079 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- Could not find support for SSE2 on this machine.
-- Could not find support for SSE3 on this machine.
-- Could not find support for SSSE3 on this machine.
-- Could not find support for SSE4.1 on this machine.
-- Could not find support for SSE4.2 on this machine.
-- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.74.0/BoostConfig.cmake (found version "1.74.0")  
-- Could not find support for SSE2 on this machine.
-- Could not find support for SSE3 on this machine.
-- Could not find support for SSSE3 on this machine.
-- Could not find support for SSE4.1 on this machine.
-- Could not find support for SSE4.2 on this machine.
-- Could not find support for SSE2 on this machine.
-- Could not find support for SSE3 on this machine.
-- Could not find support for SSSE3 on this machine.
-- Could not find support for SSE4.1 on this machine.
-- Could not find support for SSE4.2 on this machine.
-- Could NOT find TomMath (missing: TomMath_LIBRARY TomMath_INCLUDE_DIRS) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Configuring done
-- Generating done
-- Build files have been written to: /opt/zkllvm-template/build
Built target template
root@e6548116753f:/opt/zkllvm-template/build# ls 
CMakeCache.txt  CMakeFiles  CTestTestfile.cmake  Makefile  arch.c  cmake_install.cmake  libs  src
root@e6548116753f:/opt/zkllvm-template/build# 
```

**Comment by nkaskov** [comment]
It's not `zkllvm_zkllvm` target. It's `template` target.

**Comment by nkaskov** [comment]
You will get it after building the code with zkllvm.

**Comment by nkaskov** [comment]
If you are building in `build` dir, then this file is located in `build/src/`.

**Comment by BoltonBailey** [comment]
This is what I see in that folder
```
root@e6548116753f:/opt/zkllvm-template/build# ls src
CMakeFiles  CTestTestfile.cmake  Makefile  cmake_install.cmake  template.bc
```

Is it the `.bc` file?
